### PR TITLE
Fix/mpiexec mpt

### PR DESCRIPTION
**Pull request by andre-merzky**
This branches fixes the `mpiexec_mpt` behavior for Cheyenne (which is the only machine which uses that launcher).  It has been confirmed to work by the user requesting the fix.

### CODE REVIEW

File: src/radical/pilot/agent/launch_method/mpiexec.py [hunk 1]
```diff
@@ -97,24 +120,38 @@ def construct_command(self, cu, launch_script_hop):
...
+                host_string += '-host '
+                host_string += '%s -n %s ' % (','.join([node] * nslots), nslots)
+        else:
+            slots = 0
+            hosts = list()
+            for node,nslots in list(host_slots.items()):
+                slots +=          nslots
+                hosts += [node] * nslots
+            host_string += ','.join(hosts)
+            host_string += ' -n 1'
```
Comment (mtitov) [inline src/radical/pilot/agent/launch_method/mpiexec.py:142]: 1) Is variable `slots` needed here? (also variable `slots` already exists, even it is not used later, and this operation reassigns it);
2) `host_string` here doesn't need `-host`, right? (didn't find about it in the documentation..)

**Comment by mtitov** [comment]
I've added a couple of tests and besides the comments above LGTM

File: src/radical/pilot/agent/launch_method/mpiexec.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/launch_method/mpiexec.py:142]: 1: yeah, you are right - removed
2: correct, the MPT version doesn't like `-host` ...

**Comment by andre-merzky** [comment]
Thanks for adding the test cases!

**Pull request merged**